pkg/arvo/app/chat-hook.hoon

```diff
@@ -402,41 +402,15 @@
   ?>  (~(has by synced) pas)
   ::  check if read is permitted
   ?>  (is-permitted src.bol pas)
+  =/  =mailbox  (need (chat-scry pas))
+  =*  envs  envelopes.mailbox
   %-  zing
   :~  [%give %fact ~ %chat-update !>([%create pas])]~
       ?.  ?&(?=(^ backlog-start) (~(has by allow-history) pas))  ~
-      (paginate-messages pas (need (chat-scry pas)) u.backlog-start)
+      [%give %fact ~ %chat-update !>([%messages pas 0 (lent envs) envs])]~
```

Comment:
It seems unnecessary to pass both `(lent envs)` and `envs`, and it costs a traversal to calculate the length.  I think we'd be better off not passing the length around.  I might be wrong, depending on how this fits into the system at large, and it's also not really relevant to the changes in this PR.

Reply:
We use the same %messages data structure when passing on-scroll events, which do use both the start and end index.

Reply:
We could add an action / update independently for %backlog, but that seems like a premature change. I'd like to see how the network likes this change first.